test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -49,112 +55,188 @@ struct value_t {
+        return (true);
     }
 
-    std::vector<std::string>
-    get_collection_names()
+    std::shared_ptr<collection>
+    get_collection(uint64_t id)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        std::vector<std::string> collection_names;
-
-        for (auto const &it : _collections)
-            collection_names.push_back(it.first);
-
-        return (collection_names);
+        const auto &it = _collections.find(id);
+        if (it == _collections.end())
+            return nullptr;
+        /* Make a copy. */
+        return std::shared_ptr<collection>(it->second);
     }
 
-    std::map<key_value_t, key_t>
-    get_keys(const std::string &collection_name)
+    /* Not super optimal but will suffice for now. */
+    std::shared_ptr<collection>
+    get_random_collection()
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        return (_collections.at(collection_name).keys);
+        if (_collections.size() == 0)
+            return (nullptr);
+        auto it = _collections.begin();
+        std::advance(
+          it, random_generator::instance().generate_integer<uint64_t>(0, _collections.size()));
+        return std::shared_ptr<collection>(it->second);
     }
 
-    value_t
-    get_record(const std::string &collection_name, const char *key)
+    /*
+     * Retrieve the current collection count, collection names are indexed from 0 so when using this
+     * take care to avoid an off by one error.
+     */
+    uint64_t
+    get_collection_count()
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        return (_collections.at(collection_name).values.at(key));
+        return (_collections.size());
     }
 
-    void
-    insert_record(const std::string &collection_name, const char *key, const char *value)
+    /*
+     * If someone was dropping a collection concurrently this may cause them to fail, for now that
+     * is okay.
+     */
+    std::vector<std::string>
+    get_collection_names()
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.keys[key].exists = true;
-        value_t v;
-        v.value = key_value_t(value);
-        c.values.emplace(key_value_t(key), v);
+        std::vector<std::string> collection_names;
+
+        for (auto const &it : _collections)
+            collection_names.push_back(it.second->name);
+
+        return (collection_names);
     }
 
-    void
-    update_record(const std::string &collection_name, const char *key, const char *value)
+    static std::string
+    build_collection_name(const uint64_t id)
     {
-        std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.values.at(key).value = key_value_t(value);
+        return (std::string("table:collection_" + std::to_string(id)));
     }
 
     void
-    delete_record(const std::string &collection_name, const char *key)
+    add_components(timestamp_manager *tsm, workload_tracking *tracking)
```

Comment:
This method looks weird.
We could have add_component(component) instead
I understand you want to set _tsm and _tracking, they should be set in a constructor, or make those two members public and assign them directly, or create two separate setters.

Reply:
I believe if you had to do it in the constructor then you get ordering issues at the time of creating the database model and the components, two separate setters would be the way to go here.